src/main/cpp/src/get_json_object.cu

```diff
@@ -70,14 +73,52 @@ struct path_instruction {
   cudf::string_view name;
 
   // used when type is index
-  int index{-1};
+  int64_t index{-1};
 };
 
-// TODO parse JSON path
-thrust::optional<rmm::device_uvector<path_instruction>> parse_path(
-  cudf::string_scalar const& json_path)
+rmm::device_uvector<path_instruction> construct_path_commands(cudf::table_view instructions,
+                                                              rmm::cuda_stream_view stream,
+                                                              rmm::mr::device_memory_resource* mr)
 {
-  return thrust::nullopt;
+  auto const ins_types   = instructions.column(0);
+  auto const ins_names   = instructions.column(1);
+  auto const ins_indexes = instructions.column(2);
+
+  auto const s_ins_types = cudf::strings_column_view(ins_types);
+  auto const s_ins_names = cudf::strings_column_view(ins_names);
+
+  auto const d_ins_types   = cudf::column_device_view::create(s_ins_types.parent(), stream);
+  auto const d_ins_names   = cudf::column_device_view::create(s_ins_names.parent(), stream);
+  auto const d_ins_indexes = cudf::column_device_view::create(ins_indexes, stream);
+
+  rmm::device_uvector<path_instruction> path_commands(instructions.num_rows(), stream, mr);
+
+  thrust::transform(rmm::exec_policy(stream),
+                    thrust::make_counting_iterator(0),
+                    thrust::make_counting_iterator(instructions.num_rows()),
+                    path_commands.begin(),
+                    [d_types   = *d_ins_types,
+                     d_names   = *d_ins_names,
+                     d_indexes = *d_ins_indexes] __device__(auto idx) {
+                      path_instruction instruction(path_instruction_type::named);
+                      auto const type_str = d_types.element<cudf::string_view>(idx);
```

Comment:
Why don't use `path_instruction_type` but string type here?

Reply:
The path_instructions are transferred from JAVA to C++.
We can use int to represent `path_instruction_type `, like 1-> wildcard, 2-> key, ...
But these integers will be magic number in JNI repo.
Becasue the size of path_instructions is probably small, so the perf impact will be small.
There is a tradoff between readability and performance.
@ttnghia  which one do you like?

Reply:
If we have just a small number of path instructions to process then it may even be better like @revans2 said, using CPU to process them could be better. We should use the enum value only on a large dataset.